### PR TITLE
[GitHub] Add teams for ADT and Support

### PR DESCRIPTION
**Pull request by kuhar**

### CODE REVIEW

**Review by kazutakahirata — APPROVED** [review]
LGTM.  Thank you for creating these!

**Comment by kuhar** [comment]
@tstellar could you create these to teams in GitHub?

**Review by MaskRay — APPROVED** [review]

**Comment by tstellar** [comment]
Hi, we are making some changes to the notification system, could you move these paths lists into [.github/new-prs-labeler.yml](https://github.com/llvm/llvm-project/blob/main/.github/new-prs-labeler.yml). Use llvm:adt and llvm:support for the label names in the file e.g.

llvm:adt:
  - /path/1
  - /path2/

Also note that this new file uses [minimatch](https://github.com/isaacs/minimatch) glob parser which is different in some ways from the .gitignore syntax used by the CODEOWNERS file. The main difference is you need to do things like directory/** instead of directory/ to select a directory's contents.

**Comment by kuhar** [comment]
@tstellar done

File: .github/new-prs-labeler.yml [hunk 1]
```diff
@@ -479,6 +479,15 @@ llvm:analysis:
   - llvm/lib/Analysis/
   - llvm/include/llvm/Analysis/
 
+llvm:adt:
+  - llvm/include/llvm/ADT/
```
Comment (tstellar) [inline .github/new-prs-labeler.yml:483]: I think all these patterns need to end with ** .

Comment (kuhar) [reply .github/new-prs-labeler.yml:483]: Is there any way to test this before submitting?

Comment (tstellar) [reply .github/new-prs-labeler.yml:483]: Yes, see: https://github.com/llvm/llvm-project/pull/66145#discussion_r1325155972

Comment (kuhar) [reply .github/new-prs-labeler.yml]: Thanks, I tested the patterns using:
```js
const minimatch = require('minimatch')

function check_pattern(file, pattern) {
    match = minimatch(file, pattern)
    console.log(file + " : " + pattern + " : " + match)
}

check_pattern('llvm/include/llvm/ADT/Foo.cpp', 'llvm/**/ADT/*')
check_pattern('llvm/lib/ADT/Foo.cpp', 'llvm/**/ADT/*')
check_pattern('llvm/lib/unittests/ADT/Foo.cpp', 'llvm/**/ADT/*')
```

**Review by tstellar — APPROVED** [review]

**Pull request merged**